provision/kubernetes/provisioner.go

```diff
@@ -201,6 +201,9 @@ func (p *kubernetesProvisioner) InitializeCluster(c *provTypes.Cluster) error {
 }
 
 func (p *kubernetesProvisioner) ValidateCluster(c *provTypes.Cluster) error {
+	if _, ok := c.CustomData[singlePoolKey]; ok && len(c.Pools) > 1 {
```

Comment:
Checking for `len(c.Pools) != 1` would be safer because it wouldn't allow default clusters with a single pool annotation.

Reply:
Why not allow a single cluster working as single pool mode and acting as default cluster for kubernetes provisioner?

Reply:
Humm... sorry, I got it :) You're right.